### PR TITLE
Allow seed to be set to null

### PR DESCRIPTION
**Pull request by andrewjt71**
**Related Issue**
https://github.com/hautelook/AliceBundle/issues/209

The commit https://github.com/hautelook/AliceBundle/commit/4db423f3f8c9859b6db9a8ec130c3f5bae218076 changed the node type of `seed`in the bundle configuration from a scalar node to an integer node. Although the seed value is usually an integer, integer nodes can't be set to null. [The docs](https://github.com/hautelook/AliceBundle#installation) instruct users to set the seed to null in order to disable it. This is the cause of https://github.com/hautelook/AliceBundle/issues/209.

This PR reverts this change, setting `seed` back to a scalar node (scalar nodes incorporate null - [see docs](http://symfony.com/doc/current/components/config/definition.html#node-type))

### CODE REVIEW

**Comment by theofidry** [comment]
Thanks for the PR @andrewjt71!

If you do that then we must also add a validation for the value which was provided by the `integerNode` before, i.e. throw an `InvalidArgumentException` if the node value is not `null` and not an integer.

**Comment by andrewjt71** [comment]
👍  good point. Will address this

**Comment by andrewjt71** [comment]
Hi @theofidry, have added the check as requested

File: src/DependencyInjection/HautelookAliceExtension.php [hunk 1]
```diff
@@ -48,6 +48,7 @@ public function prepend(ContainerBuilder $container)
      * {@inheritdoc}
      *
      * @throws InvalidConfigurationException
+     * @throws \InvalidArgumentException
```
Comment (theofidry) [inline src/DependencyInjection/HautelookAliceExtension.php]: actually I think we can remove both throws statements, there is no point in documenting them you are not supposed to cache those kind of exceptions in the first place

File: src/DependencyInjection/HautelookAliceExtension.php [hunk 2]
```diff
@@ -61,6 +62,11 @@ public function load(array $configs, ContainerBuilder $container)
         $loader = new Loader\XmlFileLoader($container, new FileLocator(__DIR__.'/../Resources/config'));
         $loader->load('services.xml');
 
+        // Ensure seed is either an integer or null.
+        if (!(is_int($config['seed']) || is_null($config['seed']))) {
+            throw new \InvalidArgumentException('Expected the seed argument to be an integer or null. Got an argument of type ' . gettype($config['seed']) . ' instead.');
```
Comment (theofidry) [inline src/DependencyInjection/HautelookAliceExtension.php]: to be consistent with the code base, let's use a `sprintf` instead :)

**Comment by andrewjt71** [comment]
Hi @theofidry, thanks for the feedback. Have addressed your points accordingly :)

**Comment by theofidry** [comment]
Thanks @andrewjt71

**Pull request merged**